src/Stratis.Bitcoin.Features.BlockStore/BlockStoreCache.cs

```diff
@@ -34,18 +38,42 @@ public class BlockStoreCache : IBlockStoreCache
         /// <summary>Provider of time functions.</summary>
         protected readonly IDateTimeProvider dateTimeProvider;
 
-        public BlockStoreCache(IBlockRepository blockRepository, IDateTimeProvider dateTimeProvider, ILoggerFactory loggerFactory)
-            : this(blockRepository, new MemoryCache(new MemoryCacheOptions()), dateTimeProvider, loggerFactory)
-        {
-        }
+        /// <summary>The maximum amount of blocks the cache can contain.</summary>
+        public readonly int MaxCacheBlocksCount;
+
+        /// <summary>Entry options for adding blocks to the cache.</summary>
+        private readonly MemoryCacheEntryOptions blockEntryOptions;
 
-        public BlockStoreCache(IBlockRepository blockRepository, IMemoryCache memoryCache, IDateTimeProvider dateTimeProvider, ILoggerFactory loggerFactory)
+        /// <summary>Specifies amount to compact the cache by when the maximum size is exceeded.</summary>
+        /// <remarks>For example value of 0.8 will let cache remove 20% of all items when cache size is exceeded.</remarks>
+        private readonly double CompactionPercentage = 0.8;
+
+        public BlockStoreCache(
+            IBlockRepository blockRepository,
+            IDateTimeProvider dateTimeProvider,
+            ILoggerFactory loggerFactory,
+            NodeSettings nodeSettings,
+            IMemoryCache memoryCache = null)
         {
             Guard.NotNull(blockRepository, nameof(blockRepository));
-            Guard.NotNull(memoryCache, nameof(memoryCache));
 
-            this.blockRepository = blockRepository;
+            this.MaxCacheBlocksCount = nodeSettings.ConfigReader.GetOrDefault("maxCacheBlocksCount", 300);
```

Comment:
magic constant here

Reply:
It's commented
` /// <summary>The maximum amount of blocks the cache can contain.</summary>`

Reply:
i meant the `300` literal, i'd create a constant for the default value that discusses why 300

Reply:
> i'd create a constant for the default value that discusses why 300

If it's a constant it should not be configurable. And here we allow user defined value so I'm against having a constant for that. 
Though I don't mind adding an inline comment so I'll add that.